### DISCUSSION TITLE
Extending api / build chatwidget instructions

### DISCUSSION THREAD

**Discussion by Jarrodsz** [chaskiq/chaskiq#862] [Q&A] [answered]
Was just curious if it would be possible to easily extend the api ( with a plugin ) and some help on how the widget is build. We want to add a video connector and some ui modifications that probably should be contained in the plugin is that is of interest to share with the community.

for whole chaskiq the doc is very clear but not so much on how to build and extend the chat widget ui.

**Comment by michelson** [comment]
What kind of video connection? is there some API that you would like to connect to? 
if you can give more details, maybe we can help clarify.

**Reply by Jarrodsz** [reply]
This one:

https://www.100ms.live/docs/javascript/v2/guides/react-quickstart

It would be helpfull if we could build the chat widget with custom component or run it in dev mode while adding custom code.

Then export the functionality into a plugin if its possible to extend the widget with a plugin?

Can you share slme details on how to best approach?

Also is there any plans for new release anytime?

**Reply by Jarrodsz** [reply]
Ok i can look into iframe.
I see it jses webbpack and lackages but what would be an easy way to run the widget in development mode? It will be undoable to compile the assests everytime.

Do you have some docs on how to build it? The widget i mean in dev mode so it picks up codechanges straight away.

I think the plugin documentation is clear enough to start building a plugin.

**Reply by michelson** [reply, marked as answer]
you can setup an external service that responds with the proper schema to the events you setup on the panel.
the service will respond json attributes that chaskiq can understand. one of those attributes is iframe, so you can add your own js CSS if you want, and run it on development so you can make changes. the URL has to be public to chaskiq can send the messages. if you develop chaskiq on localhost you can run chaskiq on local environments too.
there is some info here: https://api-docs.chaskiq.io/docs/presentation-components
and also look on how the plugins have been done. like the whereby, or zoom, that could give you an idea on the schema required to interact on the UI with chaskiq plugin.
